worldedit-sponge/build.gradle.kts

```diff
@@ -1,54 +1,87 @@
-    setSourceCompatibility(null)
-    setTargetCompatibility(null)
+repositories {
+    mavenCentral()
 }
 
-repositories {
-    maven { url = uri("https://repo.codemc.org/repository/maven-public") }
+minecraft {
+    version("1.16.5")
+}
+
+val spongeApiVersion = "8.0.0";
+
+sponge {
+    apiVersion(spongeApiVersion)
+    license("GPL3")
+    plugin("worldedit") {
+        loader {
+            name(PluginLoaders.JAVA_PLAIN)
+            version("1.0")
+        }
+        displayName("WorldEdit")
+        version(project.ext["internalVersion"].toString())
+        entrypoint("com.sk89q.worldedit.sponge.SpongeWorldEdit")
+        description("WorldEdit is an easy-to-use in-game world editor for Minecraft, supporting both single- and multi-player.")
+        links {
+            homepage("https://enginehub.org/worldedit/")
+            source("https://github.com/EngineHub/WorldEdit")
+            issues("https://github.com/EngineHub/WorldEdit/issues")
+        }
+        contributor("EngineHub") {
+            description("Various members of the EngineHub team")
+        }
+        dependency("spongeapi") {
+            loadOrder(PluginDependency.LoadOrder.AFTER)
+            optional(false)
+        }
+    }
 }
 
 dependencies {
     api(project(":worldedit-core"))
     api(project(":worldedit-libs:sponge"))
-    api("org.spongepowered:spongeapi:7.1.0") {
+    api("org.spongepowered:spongeapi:${spongeApiVersion}") {
         exclude(group = "org.slf4j", module = "slf4j-api")
     }
     implementation(platform("org.apache.logging.log4j:log4j-bom:${Versions.LOG4J}") {
         because("Sponge 8 (will?) provides Log4J")
     })
     api("org.apache.logging.log4j:log4j-api")
-    api("org.bstats:bstats-sponge:1.7")
-    testImplementation("org.mockito:mockito-core:1.9.0-rc1")
+    // bStats isn't updated yet :(
+    api("org.bstats:bstats-sponge:2.2.1")
+    testImplementation("org.mockito:mockito-core:${Versions.MOCKITO}")
+}
+
+configure<BasePluginExtension> {
+    archivesName.set("${project.name}-api$spongeApiVersion")
 }
 
 addJarManifest(WorldEditKind.Mod, includeClasspath = true)
 
 tasks.named<ShadowJar>("shadowJar") {
     dependencies {
-        relocate ("org.bstats", "com.sk89q.worldedit.sponge.bstats") {
-            include(dependency("org.bstats:bstats-sponge:1.7"))
+        relocate("org.bstats", "com.sk89q.worldedit.sponge.bstats") {
+            include(dependency("org.bstats:"))
         }
-    }
-}
-
-if (project.hasProperty("signing")) {
-    apply(plugin = "signing")
+        include(dependency(":worldedit-core"))
 
-    configure<SigningExtension> {
-        sign("shadowJar")
-    }
-
-    tasks.named("build").configure {
-        dependsOn("signShadowJar")
+        relocate("org.antlr.v4", "com.sk89q.worldedit.sponge.antlr4")
+        include(dependency("org.antlr:antlr4-runtime"))
+        relocate("it.unimi.dsi.fastutil", "com.sk89q.worldedit.sponge.fastutil") {
```

Comment:
Does sponge really not provide fastutil?

Reply:
Would you mind opening an issue for this?

Reply:
Was done in https://github.com/SpongePowered/SpongeAPI/issues/2408.